.clang-tidy

```diff
@@ -20,12 +20,15 @@ Checks:            '*,
                     -llvm-include-order,
                     -llvmlibc-callee-namespace,
                     -llvmlibc-implementation-in-namespace,
+                    -llvmlibc-inline-function-decl,
                     -llvmlibc-restrict-system-libc-headers,
+                    -misc-include-cleaner,
+                    -misc-no-recursion,
                     -misc-non-private-member-variables-in-classes,
                     -misc-use-anonymous-namespace,
-                    -misc-no-recursion,
                     -modernize-use-nodiscard,
                     -modernize-use-trailing-return-type,
+                    -performance-avoid-endl,
```

Comment:
Some of these clang-tidy exclusions look handy, but I am sure they are a pain.  I am not going to look at every one.

Reply:
We can come back and review them later.